test/integration/auth/auth.prose.test.ts

```diff
@@ -0,0 +1,373 @@
+import { expect } from 'chai';
+import { satisfies } from 'semver';
+import * as sinon from 'sinon';
+
+import { Connection, LEGACY_HELLO_COMMAND, type MongoClient, ScramSHA256 } from '../../mongodb';
+
+function makeConnectionString(config, username, password) {
+  return `mongodb://${username}:${password}@${config.host}:${config.port}/admin?`;
+}
+
+describe('Authentication Spec Prose Tests', function () {
+  beforeEach(function () {
+    // todo(NODE-5631): fix tests to run in load balancer mode.
+    if (
+      process.env.AUTH === 'noauth' ||
+      process.env.LOAD_BALANCER ||
+      !satisfies(this.configuration.version, '>=3.7.3')
```

Comment:
Note the create user commands will fail on < 3.7.3 since `mechanisms` cannot be provided to the command.

Reply:
every test already specifies a min server version.  do we want to just move this logic into tests `requires`?  We can also use the generic predicate filter

```typescript
const metadata: MongoDBMetadataUI = {
  requires: { 
    auth: 'enabled',
    mongodb: '>=3.7.3',
    predicate: () => process.env.LOAD_BALANCER ? 'TODO(NODE-5631): fix tests to run in load balancer mode.' : true
  }
}

....

it('description', metadata, async function() { ...} );
```

Reply:
Updated to suggestion.